File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/UserGroupDeletedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;


import android.os.Parcel;
import android.support.annotation.NonNull;

import com.shareyourproxy.api.domain.model.Group;
import com.shareyourproxy.api.domain.model.User;

public class UserGroupDeletedEventCallback extends UserEventCallback {
    private final static java.lang.ClassLoader CL =
        UserGroupDeletedEventCallback.class.getClassLoader();
    public static final Creator<UserGroupDeletedEventCallback> CREATOR =
        new Creator<UserGroupDeletedEventCallback>() {
            @Override
            public UserGroupDeletedEventCallback createFromParcel(Parcel in) {
                return new UserGroupDeletedEventCallback(
                    (User) in.readValue(CL), (Group) in.readValue(CL));
            }

            @Override
            public UserGroupDeletedEventCallback[] newArray(int size) {
                return new UserGroupDeletedEventCallback[size];
            }
        };
    public final Group group;

    /**
     * Public constructor.
     *
     * @param group this events group
     */
    public UserGroupDeletedEventCallback(@NonNull User user, @NonNull Group group) {
        super(user);
        this.group = group;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(group);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/UserGroupDeletedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback


import android.os.Parcel
import android.os.Parcelable

import com.shareyourproxy.api.domain.model.Group
import com.shareyourproxy.api.domain.model.User

/**
 * A user group has been deleted.
 * @param group this events group
 */
internal final class UserGroupDeletedEventCallback(user: User, val group: Group) : UserEventCallback(user) {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(group)
    }

    companion object {
        private val CL = UserGroupDeletedEventCallback::class.java.classLoader
        val CREATOR: Parcelable.Creator<UserGroupDeletedEventCallback> = object : Parcelable.Creator<UserGroupDeletedEventCallback> {
            override fun createFromParcel(parcel: Parcel): UserGroupDeletedEventCallback {
                return UserGroupDeletedEventCallback(
                        parcel.readValue(CL) as User, parcel.readValue(CL) as Group)
            }

            override fun newArray(size: Int): Array<UserGroupDeletedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
